core/src/logging/logger.rs

```diff
@@ -1,101 +0,0 @@
-//! Abscissa logger which displays through the terminal subsystem
-// TODO(tarcieri): logfile support?
-
-use super::config::Config;
-use crate::terminal::stream::{STDERR, STDOUT};
-use log::{Level, LevelFilter, Log, Metadata, Record};
-use std::io::{Error, Write};
-use termcolor::{Color, ColorSpec, StandardStreamLock, WriteColor};
-
-/// Initialize the global logger.
-///
-/// Panics if called more than once.
-pub(super) fn init(config: &Config) {
-    log::set_boxed_logger(Box::new(Logger::new(config))).expect("error configuring global logger");
-    log::set_max_level(config.level_filter);
-}
-
-/// Abscissa component for initializing the logging subsystem
-#[derive(Debug)]
-pub struct Logger {
-    level: LevelFilter,
-}
-
-impl Logger {
-    /// Create a new logger at the given log level
-    pub fn new(config: &Config) -> Self {
-        Logger {
-            level: config.level_filter,
-        }
-    }
-
-    /// Attempt to log
-    pub fn try_log(&self, record: &Record<'_>) -> Result<(), Error> {
-        let mut stream = self.level_stream(record);
-        let now = chrono::Utc::now();
-        write!(&mut stream, "{} ", now.format("%H:%M:%S"))?;
-
-        stream.set_color(&self.level_color(record.level()))?;
-        write!(stream, "[{}] ", record.level().to_string().to_lowercase())?;
-        stream.reset()?;
```

Comment:
@tarcieri is `abscissa`'s previous formatting important enough to you that we should use a custom `tracing` formatter to mimic it?

Reply:
No, I'm fine with the default `tracing` formatter.

Separately I would *love* some kind of structured logging.

Reply:
You might be interested in https://github.com/tokio-rs/tracing/tree/master/tracing-serde ?

Reply:
> Separately I would love some kind of structured logging as an option (either JSON or key1=value1 key2=value2-formatted lines)

The default formatter is _basically_ `key1=value1`, with special handling for string fields named "message" (by convention); the next release of `tracing-subscriber` will also add a battieries-included JSON option :)

Reply:
Fantastic!